src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -371,6 +372,34 @@ public Response getDataset(@PathParam("datasetId") Integer datasetId) {
     }
   }
 
+  @GET
+  @Path("/study/{studyId}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed({ADMIN, CHAIRPERSON, DATASUBMITTER})
+  public Response getStudyById(@PathParam("studyId") Integer studyId) {
+    try {
+      Study study = datasetService.findStudyById(studyId);
+      if (Objects.isNull(study)) {
+        throw new NotFoundException("Could not find the study with id: " + studyId.toString());
+      }
+
+      // if datasets exist, get full datasets and add to study
+      Set<Integer> datasetIds = study.getDatasetIds();
+
+      if(Objects.nonNull(datasetIds)){
+        List<Integer> datasetIdList = new ArrayList<>(datasetIds);
+        List<Dataset> datasetSet = datasetService.findDatasetsByIds(datasetIdList);
+        for ( Dataset dataset : datasetSet ) {
+          study.addDataset(dataset);
+        }
+      }
```

Comment:
```suggestion
      if(Objects.isNull(datasetIds)){
        throw new NotFoundException("Study has no datasets.");
      }
      List<Integer> datasetIdList = new ArrayList<>(datasetIds);
      List<Dataset> datasetSet = datasetService.findDatasetsByIds(datasetIdList);
      for ( Dataset dataset : datasetSet ) {
        study.addDataset(dataset);
      }
```

I think studies always need to have datasets, so if `datasetIds` is null then it should error.

Reply:
Thank you!